pkg/util/containerd/containerd_util.go

```diff
@@ -35,6 +37,8 @@ var (
 
 // ContainerdItf is the interface implementing a subset of methods that leverage the Containerd api.
 type ContainerdItf interface {
+	// sts
+	GetContainers() ([]*cspec.Container, error)
```

Comment:
I think this one is not needed. The fact that the struct `ContainerdUtil` implements `GetContainers() ([]*cspec.Container, error)` already makes it meet the `ContainerUtil` requirements so we can leave this interface untouched.

Reply:
Actually, I think we need because the containerd check uses `GetContainerUtil()`, which returns an instance of `ContainerdItf`. If I don't add, our container_topology check fails.
Or I can change DD code to return `ContainerdUtil`. What do you think?

Reply:
Oh yeah I see what you mean. We'll there are different ways to do that abstraction, but I prefer to be a bit more explicit about what the function is returning. So I would go with something like:

```
func GetContainerdUtil() (*ContainerdUtil, error) {
...
}

if you want to protect yourself you can add
var _ ContainerdItf = (*ContainerdUtil)(nil) // Compile-time check
but I think it should be fine, it's referenced enough to break compile time anyway
```

The struct can always be cast "down" to match an interface, but not the other way around. In this case you get a pointer to the struct which implements the `ContainerdItf` as well as `ContainerUtil`. I noticed that CRIUtil is also implemented this way so someone probably got this wrong :smile: